WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/weapon/reload/ammo/ItemAmmo.java

```diff
@@ -156,9 +159,24 @@ public int removeAmmo(ItemStack weapon, PlayerWrapper wrapper, int amount, int m
         // and magazines in the inventory. So this reload was probably manually
         // triggered by the player, so we should use the magazines in the inventory.
         if (total == 0 && magazineSlot != -1) {
-            consumeItem(inventory, magazineSlot, inventory.getItem(magazineSlot), 1);
-            return amount;
-            // TODO refund individual bullets?
+            ItemStack item = inventory.getItem(magazineSlot);
+            int magazineAmount = Magazine.getAmmo(weapon, item);
```

Comment:
What is with this static stuff? Did you create an entirely new NBT system separate from WM?